restli-tools/src/main/java/com/linkedin/restli/tools/clientgen/fluentspec/ParameterSpec.java

```diff
@@ -50,13 +56,21 @@ public String getParamNameCaps()
 
   public ClassTemplateSpec getParamClass()
   {
-    if (_classTemplateSpec == null)
-    {
-      _classTemplateSpec = _root.classToTemplateSpec(_parameterSchema.getType());
-    }
     return _classTemplateSpec;
   }
 
+  public boolean hasParamTypeRef()
+  {
+    return !ClassUtils.getShortClassName(getParamClassName()).equals(ClassUtils.getShortClassName(
+        _declaredTypeRefClassName));
```

Comment:
Why are you comparing the short name? What if the typeref and the custom type just have different namespaces? Urn typerefs are like that.